src/client/NuGet.CommandLine.StagingExtensions/StageCommand.cs

```diff
@@ -0,0 +1,135 @@
+        {
+            Console.LogInformation(string.Format(CultureInfo.CurrentCulture, StagingResources.CreatingStageMessage, stageDisplayName));
+
+            var createResult = await stageManagementResource.Create(stageDisplayName, apiKey, Console);
+
+            Console.LogInformation(string.Format(CultureInfo.CurrentCulture, StagingResources.CreatedStageMessage, createResult.Feed));
+        }
+
+        private async Task<StageManagementResource> GetStageManagementResource(string source)
+        {
+            var sourceRepository = GetSourceRepository(new Configuration.PackageSource(source));
+            var stageManagementResourceProvider = await sourceRepository.GetResourceAsync<StageManagementResource>();
+            return stageManagementResourceProvider;
+        }
+
+        private SourceRepository GetSourceRepository(Configuration.PackageSource source)
+        {
+            var resourceProviders = new List<Lazy<INuGetResourceProvider>>();
+            resourceProviders.AddRange(FactoryExtensionsV2.GetCoreV3(Repository.Provider));
+            resourceProviders.Add(new Lazy<INuGetResourceProvider>(() => new StageManagementResourceProvider()));
+
+            return new SourceRepository(source, resourceProviders);
+        }
+
+        private string ResolveSource()
+        {
+            string source = Source;
+
+            if (string.IsNullOrEmpty(source))
+            {
+                source = SourceProvider.DefaultPushSource;
+            }
+
+            if (string.IsNullOrEmpty(source))
+            {
+                source = NuGetConstants.V3FeedUrl;
+            }
+
+            if (!string.IsNullOrEmpty(source))
+            {
+                source = SourceProvider.ResolveAndValidateSource(source);
+            }
+
+            if (string.IsNullOrEmpty(source))
+            {
+                throw new ArgumentException(NuGetResources.Error_MissingSourceParameter);
+            }
+
+            return source;
+        }
+
+        private string GetApiKey(string source)
+        {
+            string apiKey = ApiKey;
+
+            if (string.IsNullOrEmpty(apiKey))
+            {
+                apiKey = SettingsUtility.GetDecryptedValue(Settings, ConfigurationConstants.ApiKeys, source);
+            }
+
+            if (string.IsNullOrEmpty(apiKey))
+            {
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture,
+                    StagingResources.NoApiKeyFound,
+                    GetSourceDisplayName(source)));
+            }
+            return apiKey;
+        }
+
+        private static string GetSourceDisplayName(string source)
+        {
+            if (String.IsNullOrEmpty(source) || source.Equals(NuGetConstants.DefaultGalleryServerUrl, StringComparison.OrdinalIgnoreCase))
+            {
+                return StagingResources.LiveFeed + " (" + NuGetConstants.DefaultGalleryServerUrl + ")";
+            }
+
+            return "'" + source + "'";

```

Comment:
nit: some of these strings would be more readable using string interpolation 
``$"'{source}'"``
``$"{StagingResources.LiveFeed}({NuGetConstants.DefaultGalleryServerUrl})"``